### ISSUE TITLE
Auto-insert .xs/xl/as/al on REP/SEP #$10/20/30

### ISSUE THREAD

**Issue by fachat** [fachat/xa65#28]
In 65816 mode, REP and SEP modify the bits that determine the width of the A and X registers, which also affects immediate address modes. The xa currently has to be told about this with pseudo opcodes .as/.al/.xs/.xl.

xa could optionally auto-insert those when it detects corresponding REP and SEP opcodes.

**Comment by classilla** [comment]
Probably should always be an option, not the default, since it wouldn't be backwards compatible (the documentation even mentions that the pseudo-ops don't issue these instructions, so having them do so might introduce some odd bugs in source code that isn't expecting it).

**Comment by fachat** [comment]
Actually I meant it the other way round. 

If the assembler finds a SEP or REP, auto-inject the appropriate mode changes (like .xl/.xs/...)

.xs/.xl/.as/.al should indeed never inject anything, as it can be between routines etc.

**Comment by classilla** [comment]
No, I understand that. I'm concerned that's not backwards compatible either. I'm trying to think of a situation where you change the processor bit with (RS)EP but succeeding code assumes the assembler is still assembling it the old way until the pseudo-op is issued. Maybe there isn't any such situation.

**Comment by fachat** [comment]
I fully agree it should not be automatic by default, but a command line option if that is what you meant to say, Sorry if I misread.
